lightning/src/ln/inbound_payment.rs

```diff
@@ -186,6 +188,26 @@ pub fn create_from_hash(keys: &ExpandedKey, min_value_msat: Option<u64>, payment
 	Ok(construct_payment_secret(&iv_bytes, &metadata_bytes, &keys.metadata_key))
 }
 
+#[cfg(async_payments)]
+pub(super) fn create_for_spontaneous_payment(
+	keys: &ExpandedKey, min_value_msat: Option<u64>, invoice_expiry_delta_secs: u32,
+	current_time: u64, min_final_cltv_expiry_delta: Option<u16>
+) -> Result<PaymentSecret, ()> {
+	let metadata_bytes = construct_metadata_bytes(
+		min_value_msat, Method::SpontaneousPayment, invoice_expiry_delta_secs, current_time,
+		min_final_cltv_expiry_delta
+	)?;
+
+	let mut hmac = HmacEngine::<Sha256>::new(&keys.offers_base_key);
```

Comment:
I originally used a new key, but it was hard for me to tell if the new key was warranted since `offers_base_key` was already used in a couple other places and this usage felt like it was in a similar ballpark. Are there any heuristics you recommend to tell that this key "crossed the line" into over-usage?